services/bonita-classloader/bonita-classloader-impl/src/main/java/org/bonitasoft/engine/classloader/ClassLoaderServiceImpl.java

```diff
@@ -123,14 +119,13 @@ private void createClassLoader(ClassLoaderIdentifier identifier) {
         }
         VirtualClassLoader parent = getParentClassLoader(identifier);
         final VirtualClassLoader virtualClassLoader = new VirtualClassLoader(identifier.getType(), identifier.getId(), parent);
-        parent.addChild(virtualClassLoader);
         localClassLoaders.put(identifier, virtualClassLoader);
     }
 
     private VirtualClassLoader getParentClassLoader(ClassLoaderIdentifier identifier) {
         final ClassLoaderIdentifier parentIdentifier = parentClassLoaderResolver.getParentClassLoaderIdentifier(identifier);
         VirtualClassLoader parent;
-        if (parentIdentifier == null) {
+        if (ClassLoaderIdentifier.GLOBAL.equals(identifier) || parentIdentifier == null) {

```

Comment:
you are right, thanks, my tests must be bad also if it does not check this,

should i check that the resolver do not return null?